src/units/abilities.cpp

```diff
@@ -471,6 +471,50 @@ bool unit::has_ability_type(const std::string& ability) const
 	return !abilities_.child_range(ability).empty();
 }
 
+//this string abilitie is used for purpose of add a second halo if condition is matched.
+//It is put here header declaration commodity.
+const std::string unit::halo_abilities() const
+{
+	unit_ability_list halo_list = get_abilities("halo");
+	if(halo_list.empty()){
+		return "";
+	}
+	std::vector<double> priorityvec;
+	for(auto& i : halo_list) {
+		if(!(*i.ability_cfg)["priority"].empty()){
+			priorityvec.push_back((*i.ability_cfg)["priority"].to_double(0));
```

Comment:
I'm not happy with having a halo-specific priority, and won't approve this PR if it adds priorities. If both this and #7746 are merged then there would be two different priority mechanisms for halo abilities.

Reply:
i removed halo and only overlay ability remain.

Reply:
> I'm not happy with having a halo-specific priority, and won't approve this PR if it adds priorities. If both this and #7746 are merged then there would be two different priority mechanisms for halo abilities.

overwite_specials work for specials only(abilities or specials implemented) and i resolved issue with vector and map system could show 12 abilite halo maximum.

Reply:
Why do halos need a priority, and why does it need to be a different mechanism to the `[overwrite_specials]` one?

The problem is complexity of the API, which becomes more confusing if there are two things that are priorities, but which work differently.

Reply:
priority removed